### ISSUE TITLE
Trait for custom instruction decoders/encoders

### ISSUE THREAD

**Issue by ethindp** [icedland/iced#178]
Currently iced is called iced-x86 in Rust. Considering that future instruction sets may be added to iced, if they aren't currently being worked on already, would it not make sense to make a custom decoder and encoder trait for decoding and encoding custom ISAs or ISAs that iced does not yet support?

**Comment by wtfsck** [comment]
Yes that's something that could be added when it supports eg. arm/aarch64 in the future. The traits would need to be in a new (small) crate.

**Comment by ethindp** [comment]
Yeah... I was going to say that you could merge all the iced-* crates
into one but that would be rather difficult to do.

On 7/8/21, wtfsck ***@***.***> wrote:
> Yes that's something that could be added when it supports eg. arm/aarch64 in
> the future. The traits would need to be in a new (small) crate.
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/icedland/iced/issues/178#issuecomment-876245997


-- 
Signed,
Ethin D. Probst
